### PR TITLE
Adding 3DS2 passthru support to Braintree Blue

### PR DESCRIPTION
**Pull request by filipebarcos**

### CODE REVIEW

**Comment by molbrown** [comment]
Unit:
73 tests, 174 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote:
79 tests, 434 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

ECS-499

File: lib/active_merchant/billing/gateways/braintree_blue.rb [hunk 1]
```diff
@@ -651,6 +645,31 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
...
+        return if empty?(three_d_secure_opts)
+
+        parameters[:three_d_secure_pass_thru] = {
+          three_d_secure_version: three_d_secure_opts[:version],
+          eci_flag: three_d_secure_opts[:eci],
+          cavv_algorithm: three_d_secure_opts[:cavv_algorithm],
+          cavv: three_d_secure_opts[:cavv],
+          directory_response: three_d_secure_opts[:directory_response_status],
+          authentication_response: three_d_secure_opts[:authentication_response_status],
+        }.merge(xid_or_ds_trans_id(three_d_secure_opts))
```
Comment (curiousepic) [inline lib/active_merchant/billing/gateways/braintree_blue.rb]: What happens here for any blank fields?

Comment (molbrown) [reply lib/active_merchant/billing/gateways/braintree_blue.rb]: Good Q. Just tested that and interestingly, it sends the empty elements (`<directory-response nil="true"/>`). The sandbox doesn't complain, but I'll add conditionals here.

Comment (molbrown) [reply lib/active_merchant/billing/gateways/braintree_blue.rb]: Updated, including adding tests for sending only part of the `three_d_secure_opts`.

**Comment by filipebarcos** [comment]
@molbrown Do you want to create a new PR just for the sake that I can review it? 😅

**Pull request closed without merging**